pkg/virt-handler/vm.go

```diff
@@ -226,6 +189,73 @@ func MapPersistentVolumes(vm *v1.VM, restClient cache.Getter, namespace string)
 	return vmCopy, nil
 }
 
+func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, exists bool) error {
+
+	if !exists {
+		// Since the VM was not in the cache, we delete it
+		return d.domainManager.KillVM(vm)
+	} else if isWorthSyncing(vm) == false {
+		// nothing to do here.
+		return nil
+	}
+
+	// Synchronize the VM state
+	vm, err := MapPersistentVolumes(vm, d.clientset.CoreV1().RESTClient(), kubeapi.NamespaceDefault)
+	if err != nil {
+		return err
+	}
+
+	// TODO check if found VM has the same UID like the domain, if not, delete the Domain first
+
+	// Only sync if the VM is not marked as migrating. Everything except shutting down the VM is not permitted when it is migrating.
+	// TODO MigrationNodeName should be a pointer
+	if vm.Status.MigrationNodeName == "" {
+		err = d.domainManager.SyncVM(vm)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Update VM status to running
+	if vm.Status.Phase != v1.Running {
+		obj, err := kubeapi.Scheme.Copy(vm)
+		if err != nil {
+			return err
+		}
+		vm = obj.(*v1.VM)
+		vm.Status.Phase = v1.Running
+		err = d.restClient.Put().Resource("vms").Body(vm).
+			Name(vm.ObjectMeta.Name).Namespace(kubeapi.NamespaceDefault).Do().Error()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *VMHandlerDispatch) isMigrationDestination(vmName string) (bool, error) {
+
+	// If we don't have the VM in the cache, it could be that it is currently migrating to us
+	result := d.restClient.Get().Name(vmName).Resource("vms").Namespace(kubeapi.NamespaceDefault).Do()
+	if result.Error() == nil {
+		// So the VM still seems to exist
+		fetchedVM, err := result.Get()
+		if err != nil {
+			return false, err
+		}
+		if fetchedVM.(*v1.VM).Status.MigrationNodeName == d.host {
+			return true, nil
+		}
+	} else if result.Error().(*errors.StatusError).Status().Code != int32(http.StatusNotFound) {
```

Comment:
Since you are already refactoring, could you change this line to `!errors.IsNotFound(result.Error())`? You can find `errors` it in the client-go library.